src/compiler/parser.ts

```diff
@@ -5650,9 +5650,27 @@ namespace ts {
             return finishNode(node);
         }
 
+        function tryParseStringConstructorDeclaration(node: ConstructorDeclaration): ConstructorDeclaration | undefined {
+            return tryParse(() => {
+                const stringLiteral = parseLiteralNode();
+                if (isStringLiteral(stringLiteral)
+                    && stringLiteral.text === "constructor"
+                    && token() === SyntaxKind.OpenParenToken
+                ) {
+                    node.kind = SyntaxKind.Constructor;
+                    node.name = stringLiteral;

```

Comment:
Regular constructors don't have a `name` property. To avoid polymorphism you could
* explicitly set it to undefined in parseConstructorDeclaration
* set it after parseConstructorDeclarationEnd (not preferred as it still introduces a new object shape)

Reply:
True; I guess we don’t need to set that property. I was setting it just because we already have the node and it would allow you to disambiguate the exact structure of the declaration if needed, but that should be covered by `getChildren()`.